### ISSUE TITLE
Question regarding SAI and OpenNSL

### ISSUE THREAD

**Issue by davidgengenbach** [Azure/SONiC#545]
As far as I understood, all communication with the ASIC should happen with SAI.
Is the functionality to monitor individual flows, e.g. retrieving the number of bytes transferred between `10.0.0.1` and `10.0.0.2`, implemented in SAI? I could not find a real documentation of SAI itself.
Is this supported by SONiC/SAI?

Another question: can OpenNSL be used directly, i.e. through importing it directly? I did not see the library files, `libopennsl.so`, anywhere on the system, so I installed them. When calling `opennsl_driver_init(...)`, it resulted in the error message
```
Platform default configuration is used
Platform Boot flags: 0x0
insmod: ERROR: could not load module linux-kernel-bde.ko: No such file or directory
insmod: ERROR: could not load module linux-user-bde.ko: No such file or directory
open /dev/linux-user-bde: : No such file or directory
2019-12-21 11:09:53.843 (   2.423s) [main thread     ]OpenNSLStatisticsModule:219    ERR| OpenNSL: call: _return_code, opennsl message: Invalid unit
ERROR: PCI SOC device probe failed

Initializing platform
Device Configuration - SUCCESS!
Boot flags: Cold boot
No attached units.
```
basically saying the kernel modules are not loaded (and can not be found locally).
When checking through `lsmod | grep linux_kernel_bde`, sure enough the kernel modules are there and loaded. Is this a version problem? Which version is used in the most-recent ONIE Broadcom installer?

Since SONiC tries to abstract from the HW, I guess directly using OpenNSL is discouraged - but can you nevertheless give me a hint on what might create the problem? The source code of Broadcom SAI is - ohh what a surprise - closed (if I am not mistaken): https://github.com/Broadcom-Switch/SAI .
Would be greatly appreciated! Thanks!

**Comment by davidgengenbach** [comment]
This is the output when using `sudo` and copying the kernel modules, e.g. `linux-kernel-bde.ko` into the local directory because for some reason, OpenNSL searches for the kernel module in the local directory to `insmod` it... (why?):
```
Platform default configuration is used
Platform Boot flags: 0x0
insmod: ERROR: could not insert module linux-kernel-bde.ko: File exists
insmod: ERROR: could not insert module linux-user-bde.ko: File exists
open /dev/linux-user-bde: : No such file or directory
ERROR: PCI SOC device probe failed

Initializing platform
Device Configuration - SUCCESS!
Boot flags: Cold boot
No attached units.
2019-12-28 15:16:12.815 (   0.829s) [main thread     ]OpenNSLStatisticsModule:219    ERR| OpenNSL: call: _return_code, opennsl message: Invalid unit
```

The `insmod: ERROR: could not insert module linux-kernel-bde.ko: File exists` part means that the kernel module is already loaded which raises the question why the kernel module was not found by OpenNSL in the first place. 
Are there any examples of using OpenNSL directly in SONiC?

**Comment by davidgengenbach** [comment]
Push
